tocrify/api/hocr.py

```diff
@@ -207,18 +214,22 @@ def ingest_structure(self, logical):
         :param Logical logical: The logical element to be ingested. 
         """
         ingested = False
-
+        
+        # ignore leading and closing whitespace
+        label = logical.label.strip()
         # many structures are (regretfully) only labelled with 'Text'
-        if len(logical.label) > 0 and self.text and logical.label != "Text":
-            
-            begin = self.__get_best_insert_index(self.text, logical.label, 0, True)
+        if len(label) > 0 and self.text and label != "Text":
```

Comment:
`len(label) > 0` == `len(label) != 0` == `len(label)` in a condition. I find your version clearer, the pythonic way seems to be to use comparison operators as little as possible...